policy/policy.go

```diff
@@ -36,15 +36,15 @@ func Tag(policy Policy) bool {
 	return strings.HasPrefix(string(policy), "tag.")
 }
 
-func GetTagPattern(services ResourceMap, service flux.ResourceID, container string) string {
+func GetTagPattern(services ResourceMap, service flux.ResourceID, container string) Pattern {
 	if services == nil {
-		return "*"
+		return PatternAll
 	}
 	policies := services[service]
```

Comment:
> Actually I'm not sure why this doesn't NPE in the test 

aah, i think since the `(s Set) Get()` method has an implicit pointer receiver (from `map`) which makes it fine to call `Get()` on the nil map and then accessing a value of that nil map is non-panicky too (no dereferencing)

Reply:
I added the explicit checks; technically it would work fine if it were just

```
if pattern, ok := services[service].Get(TagPrefix(container)); ok {
	return NewPattern(pattern)
}
return PatternAll
```